gapir/cc/memory_manager.cpp

```diff
@@ -22,74 +22,33 @@
 #include <utility>
 #include <vector>
 
-//                 mMemory[0]
-//   ┏━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━━━━━━━━┓
-//   ┃             ┃                        ┃
-//   ┃             ┃                        ┃
-//   ┃             ┃                        ┃
-//   ┃             ┃         volatile       ┃
-//   ┃   mMemory   ┃          memory        ┃
-//   ┃             ┃                        ┃
-//   ┃             ┃                        ┨
-//   ┃             ┣━━━━━━━━━━━━━━━━━━━━━━━━┨
-//   ┃             ┃                        ┃
-//   ┃             ┃        in-memory       ┃
-//   ┃             ┃        resource        ┃
-//   ┃             ┃          cache         ┃
-//   ┃             ┃                        ┃
-//   ┗━━━━━━━━━━━━━┻━━━━━━━━━━━━━━━━━━━━━━━━┛
-//                 mMemory[mSize]
-
 namespace gapir {
 
-namespace {
-// Expected driver memory overhead to be left free as a factor of allocated
-// managed memory.
-const float kDriverOverheadFactor = 0.3f;
-}  // namespace
-
 template <typename T>
 MemoryManager::MemoryRange<T>::MemoryRange() : base(nullptr), size(0) {}
 
 template <typename T>
 MemoryManager::MemoryRange<T>::MemoryRange(T* base, uint32_t size)
     : base(base), size(size) {}
 
-MemoryManager::MemoryManager(const std::vector<uint32_t>& sizeList)
-    : mConstantMemory(nullptr, 0) {
-  for (auto size : sizeList) {
-    // Try over-allocating to leave at least (size * kDriverOverheadFactor) free
-    // bytes.
-    mSize = static_cast<uint32_t>(size * (1 + kDriverOverheadFactor));
-    mMemory.reset(new (std::nothrow) uint8_t[mSize]);
-    if (mMemory) {
-      // Free the over-allocation first, then attempt allocating the (smaller)
-      // original size.
-      mMemory.reset();
-      mSize = size;
-      mMemory.reset(new (std::nothrow) uint8_t[mSize]);
-      break;
-    }
-    GAPID_DEBUG("Failed to allocate %u bytes of volatile memory, continuing...",
-                size);
-  }
-
-  if (!mMemory) {
-    GAPID_FATAL("Couldn't allocate any volatile memory size.");
-  }
-
-  GAPID_DEBUG("Base address: %p", mMemory.get());
-  setVolatileMemory(mSize);
-}
+MemoryManager::MemoryManager()
+    : mMemory(nullptr),
+      mOpcodeMemory(nullptr, 0),
+      mConstantMemory(nullptr, 0),
+      mVolatileMemory(nullptr, 0) {}
 
 bool MemoryManager::setVolatileMemory(uint32_t size) {
-  if (size > mSize) {
-    return false;
+  mMemory.reset(new (std::nothrow) uint8_t[size]);
```

Comment:
This will likely fail with any split replay.

Reply:
Can you elaborate? I'm not sure what a split replay is!

Given my lack of understanding, it's quite likely I broke whatever this feature is!

Reply:
If you open up a Vulkan trace in the UI, I expect the report would work, but the framebuffer fetches would fail. IT basically splits a replay into 2 requests, 1 that sets up the initial state, and one that requests a particular piece of information. Importantly, it expects the volatile memory to remain untouched.